packages/react-components/src/shared/src/slots.tsx

```diff
@@ -22,7 +34,8 @@ export function getSlots(children, { _ = {}, ...slots }) {
 
         Children.forEach(children, x => {
             if (!isNil(x)) {
-                const slot = (x.props && x.props["slot"]) ?? (x.type && x.type[SLOT_KEY]);
+                const reactComponent = children as ReactElement;
+                const slot = (reactComponent.props && reactComponent.props["slot"]) ?? (reactComponent.type && (reactComponent.type as ElementTypeWithSlot)[SLOT_KEY]);
```

Comment:
I am not sure about this code you are doing:

`reactComponents.props` is the result of `children as ReactElement` but in the original code, it's `x.props` so basically this is not using the same arg as the original code.

Reply:
😱 that was bad